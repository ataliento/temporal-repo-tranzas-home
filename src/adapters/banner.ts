export interface IBanner {
  button_banner: {
    text: string;
    link: {
      id: string;
      url: string;
      linktype: string;
      fieldtype: string;
      cached_url: string;
    };
  };
  image_banner: string;
  subtitle_banner: string;
  title_banner: string;
}

import type { IBannerProps } from "@/components/Banner";

export function adapterHomeBanner(data: IBanner): IBannerProps {
  const { title_banner, subtitle_banner, image_banner, button_banner } = data;

  return {
    button: { text: button_banner.text, link: button_banner.link.url },
    image: image_banner,
    subtitle: subtitle_banner,
    title: title_banner,
  };
}
