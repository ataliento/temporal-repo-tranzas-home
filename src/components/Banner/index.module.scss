.container {
  display: flex;
  width: 100%;
  text-align: left;
  padding: var(--spacing-4) var(--spacing-5) 0 var(--spacing-5);
  background-color: var(--background-card);
  box-shadow: var(--bs-primary);
  border-radius: var(--border-radius-small);
  border: none;

  @media screen and (width <= 768px) {
    flex-flow: column-reverse nowrap;
    text-align: center;
    align-items: center;
  }
}

.title {
  font-size: var(--title-5);
  color: var(--primary);
  font-family: Ubuntu, sans-serif;
  text-align: left;
  padding-bottom: var(--spacing-3);

  @media screen and (width <= 768px) {
    max-width: 90%;
    text-align: center;
    font-size: var(--title-4);
  }
}

.subtitle {
  font-size: var(--subtitle-2);
  color: var(--gray-700);
  letter-spacing: 0;
  line-height: 1rem;
  padding-bottom: var(--spacing-4);
}

.description {
  max-width: 65%;

  @media screen and (width <= 768px) {
    max-width: 90%;
  }
}

.image {
  flex: 1;
}

.button {
  margin-bottom: var(--spacing-3);
}