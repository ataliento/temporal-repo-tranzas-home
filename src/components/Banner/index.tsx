import React from "react";
import Image from "next/image";
import { useRouter } from "next/router";
import { Button } from "@architecture-it/stylesystem";
import { Grid, Box, Typography } from "@mui/material";

import styles from "./index.module.scss";

export interface IBannerProps {
  title: string;
  subtitle: string;
  button: { text: string; link: string };
  image: string;
}

export const Banner = ({
  image,
  button = { text: "", link: "" },
  subtitle,
  title,
}: IBannerProps) => {
  const { push } = useRouter();

  return (
    <Grid container component="section">
      <Grid item className={styles.container}>
        <Box className={styles.description}>
          <Typography className={styles.title} component="h5" variant="h3">
            {title}
          </Typography>
          <Typography className={styles.subtitle} component="p" variant="body1">
            {subtitle}
          </Typography>
          <Button
            className={styles.button}
            color="primary"
            size="small"
            text={button.text}
            variant="outlined"
            onClick={() => {
              push(
                {
                  pathname: "/hacer-envio",
                  query: {
                    service: "andreaniEnvios",
                  },
                },
                "/hacer-envio"
              );
            }}
          />
        </Box>
        <Box className={styles.image}>
          <Image alt="image-banner" height={146} src={image} width={256} />
        </Box>
      </Grid>
    </Grid>
  );
};

export default Banner;
