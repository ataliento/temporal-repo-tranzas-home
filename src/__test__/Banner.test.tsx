import React from "react";
import { render } from "@testing-library/react";

import { adapterHomeBanner } from "adapters/banner";
import Banner from "@/components/Banner";

const MOCK_BANNER = {
  image_banner: "https://componentesui.blob.core.windows.net/imgs/feed_pymes/banner_feed.png",
  title_banner: "Andreani envíos",
  button_banner: {
    link: {
      id: "",
      url: "https://pymes.andreani.com/hacer-envio/andreani-env%C3%ADos?paso=origen",
      linktype: "url",
      fieldtype: "multilink",
      cached_url: "https://pymes.andreani.com/hacer-envio/andreani-env%C3%ADos?paso=origen",
    },
    text: "Ir al servicio",
  },
  subtitle_banner:
    "Con Andreani envíos podrás enviar un link a tu destinatario para que complete sus datos y abone el envío.",
};

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/status",
      pathname: "/status",
      query: "",
      asPath: "",
      back: () => {
        return "/status";
      },
    };
  },
}));

describe("<Banner/>", () => {
  it("Should render initial State without crash", () => {
    const { asFragment } = render(<Banner {...adapterHomeBanner(MOCK_BANNER)} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Should render with title", () => {
    const { getByText } = render(<Banner {...adapterHomeBanner(MOCK_BANNER)} />);

    expect(getByText(MOCK_BANNER.title_banner)).toMatchSnapshot();
  });
});
