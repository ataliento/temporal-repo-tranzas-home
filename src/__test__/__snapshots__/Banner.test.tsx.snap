// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> Should render initial State without crash 1`] = `
<DocumentFragment>
  <section
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item container css-13i4rnv-MuiGrid-root"
    >
      <div
        class="description MuiBox-root css-0"
      >
        <h5
          class="MuiTypography-root MuiTypography-h3 title css-gepadz-MuiTypography-root"
        >
          Andreani envíos
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 subtitle css-ahj2mt-MuiTypography-root"
        >
          Con Andreani envíos podrás enviar un link a tu destinatario para que complete sus datos y abone el envío.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root StyleSystem__Button_root__959dd419 MuiButton-outlined StyleSystem__Button_outlined__959dd419 MuiButton-outlinedPrimary MuiButton-sizeSmall StyleSystem__Button_sizeSmall__959dd419 MuiButton-outlinedSizeSmall MuiButton-disableElevation MuiButton-root StyleSystem__Button_root__959dd419 MuiButton-outlined StyleSystem__Button_outlined__959dd419 MuiButton-outlinedPrimary MuiButton-sizeSmall StyleSystem__Button_sizeSmall__959dd419 MuiButton-outlinedSizeSmall MuiButton-disableElevation button css-11c7338-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Ir al servicio
          </h6>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="image MuiBox-root css-0"
      >
        <img
          alt="image-banner"
          data-nimg="1"
          decoding="async"
          height="146"
          loading="lazy"
          src="https://componentesui.blob.core.windows.net/imgs/feed_pymes/banner_feed.png"
          style="color: transparent;"
          width="256"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`<Banner/> Should render with title 1`] = `
<h5
  class="MuiTypography-root MuiTypography-h3 title css-gepadz-MuiTypography-root"
>
  Andreani envíos
</h5>
`;
